x11-libs/wxGTK: Version bump to 3.2.1

listout opened the pull request:
Closes: https://bugs.gentoo.org/577030
Signed-off-by: brahmajit das <listout@protonmail.com>

jospezial commented on eclass/wxwidgets.eclass:
```diff
@@ -36,7 +36,7 @@ _WXWIDGETS_ECLASS=1
 # inheriting the eclass.  Can be either "3.0" or "3.0-gtk3".
 case ${WX_GTK_VER} in
-	3.0-gtk3) ;;
+	3.0-gtk3 | 3.2-gtk3) ;;
```
same for line 70?

listout commented:
Oh, missed that.

On Mon, Nov 21, 2022, 8:44 PM jospezial ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In eclass/wxwidgets.eclass
> <https://github.com/gentoo/gentoo/pull/28360#discussion_r1028167376>:
>
> > @@ -36,7 +36,7 @@ _WXWIDGETS_ECLASS=1
>  # The SLOT of the x11-libs/wxGTK you're targeting.  Needs to be defined before
>  # inheriting the eclass.  Can be either "3.0" or "3.0-gtk3".
>  case ${WX_GTK_VER} in
> -	3.0-gtk3) ;;
> +	3.0-gtk3 | 3.2-gtk3) ;;
>
> same for line 70?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gentoo/gentoo/pull/28360#pullrequestreview-1188424180>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP4726UOKWXNRWWCULANQGDWJOGWTANCNFSM6AAAAAASGMRLGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jospezial commented on x11-libs/wxGTK/wxGTK-3.2.1.ebuild:
```diff
@@ -0,0 +1,182 @@
+HOMEPAGE="https://wxwidgets.org/"
+SRC_URI="
+	https://github.com/wxWidgets/wxWidgets/releases/download/v${PV}/wxWidgets-${PV}.tar.bz2
+	https://dev.gentoo.org/~leio/distfiles/wxGTK-3.0.5_p20210214.tar.xz
```
old, unused?

voyageur commented on x11-libs/wxGTK/wxGTK-3.2.1.ebuild:
```diff
@@ -0,0 +1,182 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
This only inherits multilib-minimal, so we should be able to bump to EAPI 8 with this new slot?

listout replied:
Yes, sounds right.

jospezial commented:
wxGTK uses a builtin (bundled) libpcre2, called regex or pcre2. pcre2-10.37
We have libpcre2-10.40* in Gentoo.
Lets use the system lib. I found something about there:
https://github.com/wxWidgets/wxWidgets/blame/v3.2.1/configure.in#L2446-L2451
https://github.com/wxWidgets/wxWidgets/commit/d07377fdf7d619dffaacbd5ee0c4d125ca419ae8
What do you think about converting the ebuild to cmake?

leio commented:
If non-cmake still works fine, I would rather we stay with that for now, for the simple reason that I personally wouldn't be able to review a cmake conversion in a reasonable time span and would prefer we don't do everything at once - I vote we get 3.2 SLOT in and then see about such conversion after we've unblocked audacity and a few other things.

leio commented on x11-libs/wxGTK/wxGTK-3.2.1.ebuild:
```diff
@@ -0,0 +1,182 @@
+WXRELEASE=${WXVERSION%.*}-gtk3			# 3.2-gtk3
+WXRELEASE_NODOT=${WXRELEASE//./}		# 32-gtk3
+
+DESCRIPTION="GTK+ version of wxWidgets, a cross-platform C++ GUI toolkit"
```
Maybe the DESCRIPTION could be tuned a bit, but if nothing else, just GTK, not GTK+ anymore. GTK upstream dropped the `+` in the gtk4 timeline, but it's a project name so retroactive when referring it to this way.

leio commented on x11-libs/wxGTK/wxGTK-3.2.1.ebuild:
```diff
@@ -0,0 +1,182 @@
+	local myeconfargs=(
+		--with-zlib=sys
+		--with-expat=sys
+		--enable-compat28
```
This looks suspect. I think we don't need such compatibility anymore, if it's even there anymore (wxWidgets doesn't use autoconf, so it might not spit a warning when it's ineffective). `compat30` would be the thing to consider now, rather than 2.8 compat.

jospezial replied:
--enable-compat28 can even cause build errors:
https://bugs.gentoo.org/873124#c6
https://gitlab.com/kicad/code/kicad/-/issues/12665

upstream wxwidges has dropped that 2.8 compatibility code anyway in master branch:
https://github.com/wxWidgets/wxWidgets/commit/5e2da5a1066332c86f6c31223e9733337b438163
https://github.com/wxWidgets/wxWidgets/pull/22692

listout replied:
@leio Thanks updated commit.

leio commented on x11-libs/wxGTK/wxGTK-3.2.1.ebuild:
```diff
@@ -0,0 +1,182 @@
+	use doc && HTML_DOCS=( "${WORKDIR}"/wxWidgets-${WXVERSION}-docs-html/. )
+	einstalldocs
+
+	# Stray windows locale file, bug #650118
+	rm -f "${ED}"/usr/share/locale/it/LC_MESSAGES/wxmsw30-gtk3.mo || die
```
Is this actual? The `-f` option to `rm` might be hiding it not being necessary anymore (so the die is ineffective)

leio reviewed (commented):
The following review comments are by no means a real review - just some quick notes from initial skimming to push this a bit forward from my end for starters

janhenke commented:
Hi,
there has been no activity on this PR for a week now. What is needed to get this merged? It looks like all review comments have been addressed?

I am asking as I am right now in need of wxGTK 3.2 libraries for my own work. Thanks to everyone to take care of this.

listout commented:
@leio (main maintainer) is busy. We'll try to get it done by this week. Sorry for the inconvenience

janhenke commented:
I am trying out your ebuild in my private overlay right now, but the build fails at this line:
```
        # Stray windows locale file, bug #650118
        rm "${ED}"/usr/share/locale/it/LC_MESSAGES/wxmsw30-gtk3.mo || die
```

with this error:
`rm: cannot remove '/var/tmp/portage/x11-libs/wxGTK-3.2.1/image/usr/share/locale/it/LC_MESSAGES/wxmsw30-gtk3.mo': No such file or directory`

It looks like this file is no longer there and trying to delete it, should be dropped from the ebuild.

listout commented:
@janhenke thank you for pointing that. I, however, didn't face the issue/failure myself. I'll test it out again tomorrow.

janhenke commented:
One more issue, it seems the slotting is not complete, as installing the new version results in file collisions, which prevent the installation:
```

 * Messages for package x11-libs/wxGTK-3.2.1:

 * This package will overwrite one or more files that may belong to other
 * packages (see list below). You can use a command such as `portageq
 * owners / <filename>` to identify the installed package that owns a
 * file. If portageq reports that only one package owns a file then do
 * NOT file a bug report. A bug report is only useful if it identifies at
 * least two or more packages that are known to install the same file(s).
 * If a collision occurs and you can not explain where the file came from
 * then you should simply ignore the collision since there is not enough
 * information to determine if a real problem exists. Please do NOT file
 * a bug report at https://bugs.gentoo.org/ unless you report exactly
 * which two packages install the same file(s). See
 * https://wiki.gentoo.org/wiki/Knowledge_Base:Blockers for tips on how
 * to solve the problem. And once again, please do NOT file a bug report
 * unless you have completely understood the above message.
 * 
 * Detected file collision(s):
 * 
 * 	/usr/share/bakefile/presets/wx30gtk3_xrc.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_win32.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_unix.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_presets.py
 * 	/usr/share/bakefile/presets/wx30gtk3.bkl
 * 
 * Searching all installed packages for file collisions...
 * 
 * Press Ctrl-C to Stop
 * 
 * x11-libs/wxGTK-3.0.5.1:3.0-gtk3::gentoo
 * 	/usr/share/bakefile/presets/wx30gtk3.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_presets.py
 * 	/usr/share/bakefile/presets/wx30gtk3_unix.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_win32.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_xrc.bkl
 * 
 * Package 'x11-libs/wxGTK-3.2.1' NOT merged due to file collisions. If
 * necessary, refer to your elog messages for the whole content of the
 * above message.
 * 
 * The following package has failed to build, install, or execute postinst:
 * 
 *  (x11-libs/wxGTK-3.2.1:3-gtk3/3-gtk3::personal-overlay, ebuild scheduled for merge), Log file:
 *   '/var/tmp/portage/x11-libs/wxGTK-3.2.1/temp/build.log'
 * 
```

jospezial commented:
> One more issue, it seems the slotting is not complete, as installing the new version results in file collisions, which prevent the installation

I did not have any slot conflicts with https://github.com/stefantalpalaru/gentoo-overlay/blob/master/x11-libs/wxGTK/wxGTK-3.2.1-r300.ebuild

leio commented:
```rm: cannot remove '/tmp/portage/x11-libs/wxGTK-3.2.1/image/usr/share/locale/it/LC_MESSAGES/wxmsw30-gtk3.mo': No such file or directory

janhenke commented:
> ```
> rm: cannot remove '/tmp/portage/x11-libs/wxGTK-3.2.1/image/usr/share/locale/it/LC_MESSAGES/wxmsw30-gtk3.mo': No such file or directory
> ```

Ahh yes, there is an explicit remove instruction in the ebuild. From the comment it seems there was a bug which falsely left this file behind before, but it seems fixed now so that deleting it can be dropped from the ebuild.

listout commented:
@leio Noted. With @janhenke's suggestion, I've dropped the remove command for the .mo file.

listout commented:
```
* Detected file collision(s):
 * 
 * 	/usr/share/bakefile/presets/wx30gtk3_xrc.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_win32.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_unix.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_presets.py
 * 	/usr/share/bakefile/presets/wx30gtk3.bkl
 * 
 * Searching all installed packages for file collisions...
 * 
 * Press Ctrl-C to Stop
 * 
 * x11-libs/wxGTK-3.0.5.1:3.0-gtk3::gentoo
 * 	/usr/share/bakefile/presets/wx30gtk3.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_presets.py
 * 	/usr/share/bakefile/presets/wx30gtk3_unix.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_win32.bkl
 * 	/usr/share/bakefile/presets/wx30gtk3_xrc.bkl
 * 
 * Package 'x11-libs/wxGTK-3.2.1' NOT merged due to file collisions. If
 * necessary, refer to your elog messages for the whole content of the
 * above message.
```

janhenke commented:
> ```
> * Detected file collision(s):
>  * 
>  * 	/usr/share/bakefile/presets/wx30gtk3_xrc.bkl
>  * 	/usr/share/bakefile/presets/wx30gtk3_win32.bkl
>  * 	/usr/share/bakefile/presets/wx30gtk3_unix.bkl
>  * 	/usr/share/bakefile/presets/wx30gtk3_presets.py
>  * 	/usr/share/bakefile/presets/wx30gtk3.bkl
>  * 
>  * Searching all installed packages for file collisions...
>  * 
>  * Press Ctrl-C to Stop
>  * 
>  * x11-libs/wxGTK-3.0.5.1:3.0-gtk3::gentoo
>  * 	/usr/share/bakefile/presets/wx30gtk3.bkl
>  * 	/usr/share/bakefile/presets/wx30gtk3_presets.py
>  * 	/usr/share/bakefile/presets/wx30gtk3_unix.bkl
>  * 	/usr/share/bakefile/presets/wx30gtk3_win32.bkl
>  * 	/usr/share/bakefile/presets/wx30gtk3_xrc.bkl
>  * 
>  * Package 'x11-libs/wxGTK-3.2.1' NOT merged due to file collisions. If
>  * necessary, refer to your elog messages for the whole content of the
>  * above message.
> ```

I would think the files from this new slot need a wx32 in the name instead of wx30? That is the same collision I wrote about some days ago.

janhenke commented:
> right, I'm trying to figure out from where is the wx30 being suffixed from. As I've changed the slot.

There is an explicit renaming in one of the install phases.

listout commented:
> > right, I'm trying to figure out from where is the wx30 being suffixed from. As I've changed the slot.
> 
> There is an explicit renaming in one of the install phases.

yep, saw that, changed it, and building again.

janhenke commented on x11-libs/wxGTK/wxGTK-3.2.1.ebuild:
```diff
@@ -0,0 +1,177 @@
+	pushd "${ED}"/usr/share/bakefile/presets >/dev/null || die
+	local f
+	for f in wx*; do
+		mv "${f}" "${f/wx/wx30gtk3}" || die
```
Change this to wx32gtk3? Or just wx32?

listout replied:
Right. Will do that in next force push

listout commented:
> > right, I'm trying to figure out from where is the wx30 being suffixed from. As I've changed the slot.
> 
> There is an explicit renaming in one of the install phases.

I tried this diff still wx30gtk3 files present in bakefile folder

```diff
diff --git a/x11-libs/wxGTK/wxGTK-3.2.1.ebuild b/x11-libs/wxGTK/wxGTK-3.2.1.ebuild
index 6394221de5e..bf2d8a8ce11 100644
--- a/x11-libs/wxGTK/wxGTK-3.2.1.ebuild
+++ b/x11-libs/wxGTK/wxGTK-3.2.1.ebuild
@@ -84,7 +84,7 @@ src_prepare() {
 		-e "s:\(WX_VERSION=\).*:\1${WXVERSION}:" \
 		-e "s:\(WX_RELEASE=\).*:\1${WXRELEASE}:" \
 		-e "s:\(WX_SUBVERSION=\).*:\1${WXSUBVERSION}:" \
-		-e '/WX_VERSION_TAG=/ s:${WX_RELEASE}:3.0:' \
+		-e '/WX_VERSION_TAG=/ s:${WX_RELEASE}:3.2:' \
 		configure || die
 }
```

listout commented:
@janhenke I think the file collision issue should be fixed now, thanks for the help.
